### PR TITLE
Inserter upside down rotate fix.

### PR DESCRIPTION
**Pull request by camerongillette**
Changes
-Logic to handle mirroring by adding additional/optional parameter to placeable variable
-Code cleanup
-Logging cleanup
-Wrote basic documentation on contents of placeable array contents

Testing Instructions
-Add inserter of any type and rotate. The inserter should know rotate instead of flip upside down.

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -28,6 +28,15 @@ function GETfromUrl(){
...
+/*
+placeable[0] = image path
+placeable[1] = rotation
+placeable[2] = direction
+placeable[3] = width
+placeable[4] = height/length
+placeable[5] = mirror flipped horizontal rotation
+*/
+
 var placeable = [
```
Comment (Zeragamba) [inline js/script.js:41]: We should break this out into a json file later.

File: js/script.js [hunk 2]
```diff
@@ -310,10 +319,14 @@ function rotatePreview() {
     if (preview != null && preview.dataset.r != 0) {
         var direction = (Number(preview.dataset.direction) + 2) % 8;
         var dirStart = Number(preview.dataset.dirstart);
-        console.log('sadjkhkjdhs' + dirStart);
+        console.log('Direction Start' + dirStart);
```
Comment (Zeragamba) [inline js/script.js]: I think we should remove debugging console logs from the code.

File: js/script.js [hunk 3]
```diff
@@ -444,7 +468,7 @@ function setPreviewFollow(mouseLoc){
 }
 
 function itemClick() {
-    createPreview(this.dataset.url, this.dataset.r, this.dataset.direction, this.dataset.w, this.dataset.h);
+    createPreview(this.dataset.url, this.dataset.r, this.dataset.direction, this.dataset.w, this.dataset.h, this.dataset.mirrorFlippedHorizontal);
```
Comment (Zeragamba) [inline js/script.js]: Seems like this method uses a lot of values from dataset. It might be easier to pass the dataset object it's self

**Review by Zeragamba — CHANGES_REQUESTED** [review]

**Comment by camerongillette** [comment]
@SpyMaster356 Ready for re-review

**Comment by Zeragamba** [comment]
@camerongillette can you push a new commit (edit the README or something) to trigger a Travis build?

**Review by Zeragamba — APPROVED** [review]

**Comment by Zeragamba** [comment]
actually, looks like closing and reopening works a well

File: js/script.js [hunk 4]
```diff
@@ -482,7 +508,7 @@ function tileClick() {
             y -= 1;
         }
         document.querySelector('[data-x="' + x + '"][data-y="' + y + '"]').click();
-    } else if (!isBlocked(previewdiv.dataset.name, this.dataset.x, this.dataset.y)) {
+    } else if (!isBlocked(previewdiv.dataset.name, this.dataset.x, this.dataset.y, )) {
```
Comment (Zeragamba) [inline js/script.js]: Got an extra tailing comma here

**Review by Zeragamba — CHANGES_REQUESTED** [review]

**Comment by camerongillette** [comment]
@SpyMaster356 Ready for re-review.

**Review by Zeragamba — APPROVED** [review]

**Pull request merged**